pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -217,38 +217,43 @@ func (r *ReconcileDNSZone) reconcileDNSProvider(actuator Actuator, dnsZone *hive
 	r.logger.Debug("Retrieving current state")
 	err := actuator.Refresh()
 	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "AccessDeniedException" {
-			conds, changed := controllerutils.SetDNSZoneConditionWithChangeCheck(
-				dnsZone.Status.Conditions,
-				hivev1.InsufficientCredentialsCondition,
-				corev1.ConditionTrue,
-				accessDeniedReason,
-				awsErr.Message(),
-				controllerutils.UpdateConditionIfReasonOrMessageChange,
-			)
-			if changed {
-				dnsZone.Status.Conditions = conds
-				if err := r.Status().Update(context.Background(), dnsZone); err != nil {
-					return reconcile.Result{}, err
-				}
+		conditionsChanged := actuator.SetConditionsForError(err)
+
+		if conditionsChanged {
+			if err := r.Status().Update(context.Background(), dnsZone); err != nil {
+				return reconcile.Result{}, err
 			}
-			r.logger.WithError(err).Warn("Credentials are lacking permissions")
-			return reconcile.Result{}, err
 		}
+
 		r.logger.WithError(err).Error("Failed to retrieve hosted zone and corresponding tags")
 		return reconcile.Result{}, err
 	}
 
-	conds, changed := controllerutils.SetDNSZoneConditionWithChangeCheck(
+	accessDeniedConds, accessDeniedCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
```

Comment:
These conditions will only get set to false when there are no longer **any** errors. For example, let's say that there was initially an authentication error. The user fixes that but then there is an access denied error. The DNSZone will show that there is still an authentication error and that there is now **also** an access denied error.

Reply:
In your example, wouldn't fixing the authentication error cause authenticationFailureCondsChanged to become true?

I'm not seeing how the code works the way you describe.

Reply:
Ah, I get it. You're saying because we'd go into the ```if err != nil``` above and so this code wouldn't be reached. Good catch.